Internal Server Error occurs when answering challenges where dynamic scoring is set.

```
[2018-05-08 01:42:50,728] ERROR in app: Exception on /chal/1 [POST]
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 2284, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1807, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1710, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1805, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1791, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/opt/CTFd/CTFd/utils/decorators.py", line 15, in during_ctf_time_only_wrapper
    return f(*args, **kwargs)
  File "/opt/CTFd/CTFd/utils/decorators.py", line 56, in viewable_without_authentication_wrapper
    return f(*args, **kwargs)
  File "/opt/CTFd/CTFd/challenges.py", line 394, in chal
    status, message = chal_class.attempt(chal, request)
  File "/opt/CTFd/CTFd/plugins/DynamicValueChallenge/__init__.py", line 160, in attempt
    if get_key_class(chal_key.type).compare(chal_key.flag, provided_key):
  File "/opt/CTFd/CTFd/plugins/keys/__init__.py", line 28, in compare
    saved = chal_key_obj.flag
AttributeError: 'unicode' object has no attribute 'flag'
[2018-05-08 01:42:50 +0000] [34] [ERROR] Error handling request /chal/1
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/async.py", line 56, in handle
    self.handle_request(listener_name, req, client, addr)
  File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/ggevent.py", line 152, in handle_request
    super(GeventWorker, self).handle_request(*args)
  File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/async.py", line 107, in handle_request
    respiter = self.wsgi(environ, resp.start_response)
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 2301, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 2287, in wsgi_app
    response = self.handle_exception(e)
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1740, in handle_exception
    return self.finalize_request(handler(e), from_error_handler=True)
  File "/opt/CTFd/CTFd/utils/__init__.py", line 137, in general_error
    return render_template('errors/500.html', error=error.description), 500
AttributeError: 'exceptions.AttributeError' object has no attribute 'description'
```

--- a/__init__.py
+++ b/__init__.py
@@ -157,7 +157,7 @@ class DynamicValueChallenge(BaseChallenge):
         provided_key = request.form['key'].strip()
         chal_keys = Keys.query.filter_by(chal=chal.id).all()
         for chal_key in chal_keys:
-            if get_key_class(chal_key.type).compare(chal_key.flag, provided_key):
+            if get_key_class(chal_key.type).compare(chal_key, provided_key):
                 return True, 'Correct'
         return False, 'Incorrect'
 

